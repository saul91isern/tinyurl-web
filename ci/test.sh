#!/bin/sh

echo "----- Install dependencies -----"
yarn
echo "----- Linter -----"
yarn test --ci

