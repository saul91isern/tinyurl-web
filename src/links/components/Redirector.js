import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import { clearRedirect } from "../routines";

export const Redirector = ({ redirect, clearRedirect }) => {
  useEffect(() => {
    clearRedirect();
  }, [clearRedirect]);
  return <Redirect to={{ pathname: redirect }} />;
};

Redirector.propTypes = {
  redirect: PropTypes.string,
  clearRedirect: PropTypes.func,
};

export default connect(null, { clearRedirect })(Redirector);
