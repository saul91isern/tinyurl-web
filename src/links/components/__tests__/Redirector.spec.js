/* eslint-disable quotes */
import React from "react";
import { render } from "../../../test/render";
import { Redirector } from "../Redirector";

describe("<Redirector />", () => {
  it("renders Redirector", async () => {
    const clearRedirect = jest.fn();
    const { container } = render(
      <Redirector redirect={"foo"} clearRedirect={clearRedirect} />,
      {}
    );
    expect(container).toMatchSnapshot();
  });
});
