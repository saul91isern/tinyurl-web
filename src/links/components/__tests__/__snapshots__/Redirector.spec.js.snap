// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Redirector /> renders Redirector 1`] = `<div />`;
